Content.Server/NPC/Pathfinding/PathfindingSystem.cs

```diff
@@ -268,11 +269,12 @@ public async Task<PathResultEvent> GetRandomPath(
                 return new PathResultEvent(PathResult.NoPath, new List<PathPoly>());
 
             var layer = 0;
-            var mask = 0;
+            var mask = GetCollisionMask(entity);
 
             if (TryComp<FixturesComponent>(entity, out var fixtures))
             {
-                (layer, mask) = _physics.GetHardCollision(entity, fixtures);
+                (layer, var newmask) = _physics.GetHardCollision(entity, fixtures);
+                mask |= newmask;
             }
```

Comment:
This bulldozes the difference between flying and ground mobs.

Reply:
This just adds BotImpassible if it's on the blackboard